### PR TITLE
Adding components props types to storybook's Meta type generic argument

### PR DESCRIPTION
**Pull request by felipe-jm**
Since the type Meta from Storybook accepts a generic argument to type the "args" property, I think it would be a nice improvement to use it and improve the types in components stories.

### CODE REVIEW

**Comment by willianjusten** [comment]
When you type the `Story`  using `Story<type>` you're already defining the type of the `args`, so you don't need to type `Meta`. Doing twice is kinda of redundant, that's why it's not necessary, but thanks for trying to help =D

I noticed that on `Button` we didn't type correctly, so to make this PR still valid, can you remove from `Meta` and let only the types on `Button`? I'll comment the lines that should stay.

**Comment by marcoswebmasteroficial** [comment]
> Since the type Meta from Storybook accepts a generic argument to type the "args" property, I think it would be a nice improvement to use it and improve the types in components stories.

https://github.com/storybookjs/design-system

File: src/components/Banner/stories.tsx [hunk 1]
```diff
@@ -19,7 +19,7 @@ export default {
   parameters: {
     layout: 'fullscreen'
   }
-} as Meta
+} as Meta<BannerProps>
```
Comment (willianjusten) [inline src/components/Banner/stories.tsx]: All of that can be removed, as I said, we already defined the type inside `Story` which is more readable.

File: src/components/Button/stories.tsx [hunk 2]
```diff
@@ -13,23 +13,23 @@ export default {
       type: ''
     }
   }
-} as Meta
+} as Meta<ButtonProps>
 
-export const Default: Story = (args) => <Button {...args} />
+export const Default: Story<ButtonProps> = (args) => <Button {...args} />
```
Comment (willianjusten) [inline src/components/Button/stories.tsx:18]: This one is indeed useful and correct, keep those please.

File: src/components/Button/stories.tsx [hunk 3]
```diff
@@ -13,23 +13,23 @@ export default {
...
 
 Default.args = {
   children: 'Buy now'
 }
 
-export const withIcon: Story = (args) => <Button {...args} />
+export const withIcon: Story<ButtonProps> = (args) => <Button {...args} />
```
Comment (willianjusten) [inline src/components/Button/stories.tsx:24]: And here.

File: src/components/Button/stories.tsx [hunk 4]
```diff
@@ -13,23 +13,23 @@ export default {
...
 
 withIcon.args = {
   size: 'small',
   children: 'Buy now',
   icon: <AddShoppingCart />
 }
 
-export const asLink: Story = (args) => <Button {...args} />
+export const asLink: Story<ButtonProps> = (args) => <Button {...args} />
```
Comment (willianjusten) [inline src/components/Button/stories.tsx:32]: And here.

File: src/components/Button/stories.tsx [hunk 5]
```diff
@@ -1,6 +1,6 @@
 import { Story, Meta } from '@storybook/react/types-6-0'
 import { AddShoppingCart } from '@styled-icons/material-outlined/AddShoppingCart'
-import Button from '.'
+import Button, { ButtonProps } from '.'
```
Comment (willianjusten) [inline src/components/Button/stories.tsx:3]: Here also to have the type.

File: src/components/Button/stories.tsx [hunk 6]
```diff
@@ -13,23 +13,23 @@ export default {
       type: ''
     }
   }
-} as Meta
+} as Meta<ButtonProps>
```
Comment (willianjusten) [inline src/components/Button/stories.tsx]: This one can be removed as the others.

**Review by willianjusten — CHANGES_REQUESTED** [review]
Please, remove the types on `Meta` and keep just the types on `Button` as I pointed.

**Comment by felipe-jm** [comment]
> Please, remove the types on `Meta` and keep just the types on `Button` as I pointed.

Done! I'm happy I could help a little. Thank you! :blush:

**Pull request merged**